send.c

```diff
@@ -571,8 +571,16 @@ static struct safe_buffer_list *add_ra_options_route(struct safe_buffer_list *sb
 		memset(&rinfo, 0, sizeof(rinfo));
 
 		rinfo.nd_opt_ri_type = ND_OPT_ROUTE_INFORMATION;
-		/* XXX: the prefixes are allowed to be sent in smaller chunks as well */
-		rinfo.nd_opt_ri_len = 3;
+		if (route->PrefixLen == 0) {
+			rinfo.nd_opt_ri_len = 1;
+		} else if (route->PrefixLen > 0 && route->PrefixLen <= 64) {
+			rinfo.nd_opt_ri_len = 2;
+		} else if (route->PrefixLen > 64 && route->PrefixLen <= 128) {
+			rinfo.nd_opt_ri_len = 3;
+		} else {
+			route = route->next;
+			continue;
+		}
```

Comment:
silently dropping the route here seems weird. At the very least it should drop; but also it should be IMPOSSIBLE to reach this if we validate routes earlier.

Reply:
@y-kzm  bump